sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -272,5 +260,19 @@ object SharedState extends Logging {
       sparkWarehouseDir
     }
     logInfo(s"Warehouse path is '$warehousePath'.")
+    val confClone = sparkConf.clone()
```

Comment:
This change add a unnecessary overhead in the `SparkSQLCLIDriver` side? https://github.com/apache/spark/blob/ba974ea8e4cc8075056682c2badab5ca64b90047/sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala#L136

Reply:
Or We can return initialConfigs -- Seq(WAREHOUSE_PATH.key, hiveWarehouseKey) instead?

Reply:
We can skip all these if `initialConfigs` is empty?

Reply:
> We can skip all these if `initialConfigs` is empty?

A bit tricky， look we need to clone or not based `initialConfigs`